### PR TITLE
add avahi Advertiser to UI

### PR DESCRIPTION
**Pull request by donavanbecker**
add avahi Advertiser to Homebridge UI for Homebridge v1.4.0-beta

### CODE REVIEW

File: src/modules/plugins/plugins.service.ts [hunk 1]
```diff
@@ -581,6 +581,22 @@ export class PluginsService {
...
+          const config: HomebridgeConfig = await fs.readJson(this.configService.configPath);
+          config.bridge.advertiser = 'avahi';
+          await fs.writeJsonSync(this.configService.configPath, config);
+        } catch (e) {
+          this.logger.warn('Could not update config.json', e.message);
+        }
+      }
+    }
+    // end 1.3.x -> 1.4.0 upgrade
+
```
Comment (oznu) [inline src/modules/plugins/plugins.service.ts:599]: I don't think we need to change the current advertiser for existing users when they upgrade. It's it's "working" we should leave existing installs alone.

File: ui/src/app/modules/settings/settings.component.html [hunk 2]
```diff
@@ -45,6 +45,7 @@ <h5 class="primary-text mt-3" translate="settings.network.title_network"></h5>
         <select class="custom-select" [formControl]="legacyMdnsFormControl"
           [attr.aria-label]=" 'settings.mdns_advertiser' | translate">
           <option value="bonjour-hap">Bonjour HAP</option>
+          <option value="avahi">Avahi</option>
```
Comment (oznu) [inline ui/src/app/modules/settings/settings.component.html:48]: We could only display this option if the user is running Linux.

Comment (oznu) [reply ui/src/app/modules/settings/settings.component.html:48]: Maybe having a way for the UI to check if Avahi is available would be useful, as when running Docker the "avahi" service is only started on the Ubuntu variant of the image by default.

File: src/modules/plugins/plugins.service.ts (hunk 1 above)
Comment (donavanbecker) [reply src/modules/plugins/plugins.service.ts:599]: Okay! Would you like me to remove this?

I only changed this for Linux users.

File: ui/src/app/modules/settings/settings.component.html (hunk 2 above)
Comment (donavanbecker) [reply ui/src/app/modules/settings/settings.component.html:48]: Sounds good!

File: src/modules/plugins/plugins.service.ts (hunk 1 above)
Comment (donavanbecker) [reply src/modules/plugins/plugins.service.ts]: I have removed this.

File: ui/src/app/modules/settings/settings.component.html (hunk 2 above)
Comment (Supereg) [reply ui/src/app/modules/settings/settings.component.html:48]: Can we interface with homebridge to report which advertiser was picked in "choose default" mode? see https://github.com/homebridge/HAP-NodeJS/pull/922

**Comment by donavanbecker** [comment]
@oznu, any other changes needed to get this merged? and into a beta?

**Pull request merged**